stan-dev/math
# Feature/issue 179 robust integrator

#### Summary:

Change the dense_output scheme of odeint to the controlled error stepper scheme. Also increase the dt step size to 1 as per recommendation of odeint's controlled error steppers this initial step size should be larger than the typical step size (and the unit 1 should be large wrt to the precision needed).

#### Intended Effect:

The dense_output scheme can freeze whenever a set of parameter values are tried which result in a too stiff ODE problem. The controlled error stepper expects an initial dt which is adjusted according to the relative and absolute errors requested. The controlled error stepper tries 1000 times to find an appropriate step size and throws an exception if this is not possible within 1000 attempts.

This makes the ODE integrator robust against extreme parameter draws which occur often during warmup.

#### How to Verify:

This fixes ODE lockups as documented in issue #1510 (stan repo). Run ode tests to verify that ODE integrator still works as before.

#### Side Effects:

None.

#### Documentation:

None.

#### Reviewer Suggestions:

Michael, Daniel or Bob
